continuous_integration/flake8_diff.sh

```diff
@@ -0,0 +1,35 @@
+#!/bin/sh
+
+set -e
+
+# Travis goes the git clone with a limited depth (50 at the time of
+# writing). This may not be enough to find the common ancestor with
+# $REMOTE/master so we unshallow the git checkout
+git fetch --unshallow || echo "Unshallowing the git checkout failed"
+
+# Tackle both common cases of origin and upstream as remote
+# Note: upstream has priority if it exists
+git remote -v
+git remote | grep upstream && REMOTE=upstream || REMOTE=origin
+git fetch -v $REMOTE master:remote_master
```

Comment:
Someone with more git-fu than me may know this. I couldn't figure out a way to get the equivalent on $REMOTE/master without checking a branch locally ...
The problem comes from the fact that Travis checks out a single branch like this:
```
git clone --depth=50 --branch=test https://github.com/lesteve/nilearn.git lesteve/nilearn
```

So even doing `git fetch origin` upstream/master, origin/master is not going to show up in `git branch -r`.